SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestInitialOnly/SpatialTestInitialOnlyForInterestActor.cpp

```diff
@@ -115,10 +115,11 @@ void ASpatialTestInitialOnlyForInterestActor::PrepareTest()
 	AddStep(TEXT("Server change value."), FWorkerDefinition::Server(1), nullptr, [this]() {
 		TArray<AActor*> SpawnActors;
 		UGameplayStatics::GetAllActorsOfClass(GetWorld(), ASpatialTestInitialOnlySpawnActor::StaticClass(), SpawnActors);
+		AssertEqual_Int(SpawnActors.Num(), 1, TEXT("There should be exactly one InitialOnly actor in the world."));
 		for (AActor* Actor : SpawnActors)
 		{
 			ASpatialTestInitialOnlySpawnActor* SpawnActor = Cast<ASpatialTestInitialOnlySpawnActor>(Actor);
-			if (SpawnActor != nullptr)
+			if (AssertIsValid(SpawnActor, TEXT("SpawnActor should be valid.")))
```

Comment:
Did this actual fail? Or just a driveby?

Reply:
I think this is an actual fail here. `SpawnActor` should have been created in the previous steps.

Reply:
I think Mike was asking whether I have seen it fail in my testing, and I haven't.
But I think my philosophy for writing tests is that for every `if`, I ask what I want the test to do if that condition is unsatisified: in this case, if the SpawnActor is not valid, the test would previously proceed to finish step and probably fail at some point further down. However, not having the SpawnActor is pretty bad thing and I would instead expect the test to fail, so I try replacing flow-control (where possible) with assert-and-flow-control.